float64[3] initial
float64[3] goal
float64[3] line_start
float64[3] line_end
